### ISSUE TITLE
Indirect call signature mismatch SDL2_Mixer

### ISSUE THREAD

**Issue by Oipo** [emscripten-core/emscripten#10111]
I didn't encounter this before and I'm unable to track down what is causing this. I do know that `-s LEGALIZE_JS_FFI=0` does not resolve this error. I also remove the ~/.emscripten_cache and ~/.emscripten_ports directories, to no avail.

The console shows the following error:

```
exception thrown: RuntimeError: indirect call signature mismatch,Mix_LoadMUSType_RW@https://www.realmofaesir.com/ibh_frontend.js line 1897 > WebAssembly.instantiate:wasm-function[9737]:0x25c1e4
Mix_LoadMUS@https://www.realmofaesir.com/ibh_frontend.js line 1897 > WebAssembly.instantiate:wasm-function[9735]:0x25bfe6
main@https://www.realmofaesir.com/ibh_frontend.js line 1897 > WebAssembly.instantiate:wasm-function[6105]:0xf0848
```

**Comment by Oipo** [comment]
I get a similar error when running tests/sdl2_mixer.c in the browser:

```
exception thrown: RuntimeError: indirect call signature mismatch,Mix_LoadWAV_RW@https://www.realmofaesir.com/test.js line 1873 > WebAssembly.instantiate:wasm-function[255]:0x4564
main@https://www.realmofaesir.com/test.js line 1873 > WebAssembly.instantiate:wasm-function[243]:0x30bd
Module._main@https://www.realmofaesir.com/test.js:9201:32
callMain@https://www.realmofaesir.com/test.js:9627:28
doRun@https://www.realmofaesir.com/test.js:9689:31
run/<@https://www.realmofaesir.com/test.js:9700:7
```

I compiled the test with the following command:
```
em++ -g3 -ggdb -O0 -s USE_SDL_MIXER=2 -s USE_OGG=1 -s USE_VORBIS=1 sdl2_mixer.c -o test.html --no-heap-copy --preload-file sounds/pluck.ogg@/sound.ogg
```

**Comment by Oipo** [comment]
Fixed in #10237
